auto-value-gson-extension/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -961,6 +998,33 @@ private MethodSpec createReadMethod(ClassName className,
     return readMethod.build();
   }
 
+  private Property findUnrecognizedJsonPropertiesContainer(List<Property> properties, ProcessingEnvironment processingEnvironment) {
+    List<Property> unrecognisedProperties = properties.stream()
+      .filter(this::isUnrecognizedJsonPropertiesContainer)
+      .collect(Collectors.toList());
+    if (unrecognisedProperties.size() > 1) {
+      processingEnvironment.getMessager().printMessage(
+        Diagnostic.Kind.ERROR,
+        String.format("Only one method can be annotated with %s", UnrecognizedJsonProperties.class.getSimpleName()),
+        unrecognisedProperties.get(1).element
+      );
+      return unrecognisedProperties.get(0);
```

Comment:
I didn't dig deep in this questions, but here's what I've learned about annotation processors so far:

The doc recommends to preffer  
> processor should throw an uncaught exception only in situations where no error recovery or reporting is feasible.

Reply:
I didn't dig deep in this questions, but here's what I've learned about annotation processors so far:

[The doc](https://docs.oracle.com/javase/6/docs/api/javax/annotation/processing/Processor.html) recommends to prefer error reporting to uncaught exceptions. 
> processor should throw an uncaught exception only in situations where no error recovery or reporting is feasible.

I guess it's because users would like to see an error like "Only one method can be annotated with ..." rather than "Annotation processor X failed with an exception".

I also noticed that this processor doesn't throw exceptions in case of other errors. It logs error(logged error fails compilation for user) and carries on with generation. I decided to just stick to existing approach and don't spend time for digging deeper. Any way, if you find this question very important I'm ready to dig deeper and learn more about best practises in annotation processing.